scipy/stats/tests/test_stats.py

```diff
@@ -276,7 +280,96 @@ def test_basic(self):
         b = array([0, 0, 3])
         r, prob = stats.pearsonr(a, b)
         assert_approx_equal(r, np.sqrt(3)/2)
-        assert_approx_equal(prob, 1.0/3)
+        assert_approx_equal(prob, 1/3)
+
+    def test_constant_input(self):
+        # Zero variance input
+        # See https://github.com/scipy/scipy/issues/3728
+        with assert_warns(stats.PearsonRConstantInputWarning):
+            r, p = stats.pearsonr([0.667, 0.667, 0.667], [0.123, 0.456, 0.789])
+            assert_equal(r, np.nan)
+            assert_equal(p, np.nan)
+
+    def test_near_constant_input(self):
+        # Near constant input (but not constant):
+        x = [2, 2, 2 + np.spacing(2)]
+        y = [3, 3, 3 + 6*np.spacing(3)]
+        with assert_warns(stats.PearsonRNearConstantInputWarning):
+            # r and p are garbage, so don't bother checking them in this case.
+            # (The exact value of r would be 1.)
+            r, p = stats.pearsonr(x, y)
+
+    def test_very_small_input_values(self):
+        # Very small values in an input.  A naive implementation will
+        # suffer from underflow.
+        # See https://github.com/scipy/scipy/issues/9353
+        x = [0.004434375, 0.004756007, 0.003911996, 0.0038005, 0.003409971]
+        y = [2.48e-188, 7.41e-181, 4.09e-208, 2.08e-223, 2.66e-245]
+        r, p = stats.pearsonr(x,y)
+
+        # The expected values were computed using mpmath with 80 digits
+        # of precision.
+        assert_allclose(r, 0.7272930540750450)
+        assert_allclose(p, 0.1637805429533202)
+
+    def test_very_large_input_values(self):
+        # Very large values in an input.  A naive implementation will
+        # suffer from overflow.
+        # See https://github.com/scipy/scipy/issues/8980
+        x = 1e90*np.array([0, 0, 0, 1, 1, 1, 1])
+        y = 1e90*np.arange(7)
+
+        r, p = stats.pearsonr(x, y)
+
+        # The expected values were computed using mpmath with 80 digits
+        # of precision.
+        assert_allclose(r, 0.8660254037844386)
+        assert_allclose(p, 0.011724811003954638)
+
+    def _test_extremely_large_input_values(self):
```

Comment:
Why is there a leading underscore in this test name? Probably a typo.

If you remove that, you can verify locally that `python runtests.py --mode=full -t "scipy/stats/tests/test_stats.py::TestCorrPearsonr"` will execute one additional test.

Reply:
I don't know why I put a leading underscore in the name, and I'm embarrassed that I didn't notice that the test wasn't being run 😟 .  I fixed it in my latest update.